src/dotnet/APIView/APIViewWeb/Repositories/CosmosReviewRepository.cs

```diff
@@ -108,5 +111,126 @@ public async Task<IEnumerable<ReviewModel>> GetReviewsAsync(string serviceName,
+            List<string> authors, List<string> tags, bool? isClosed, List<int> filterTypes, bool? isApproved, int offset, int limit)
+        {
+            (IEnumerable<ReviewModel> Reviews, int TotalCount) result = (Reviews: new List<ReviewModel>(), TotalCount: 0);
+
+            // Build up Query
+            var queryStringBuilder = new StringBuilder("SELECT * FROM Reviews r");
+            queryStringBuilder.Append(" WHERE IS_DEFINED(r.id)"); // Allows for appending the other query parts in any order
+
+            if (packageNames != null && packageNames.Count > 0)
+            {
+                var packageNamesAsQueryStr = ArrayToQueryString<string>(packageNames);
+                queryStringBuilder.Append($" AND r.PackageDisplayName IN {packageNamesAsQueryStr}");
+            }
+
+            if (languages != null && languages.Count > 0)
+            {
+                var languagesAsQueryStr = ArrayToQueryString<string>(languages);
+                queryStringBuilder.Append($" AND r.Revisions[0].Files[0].Language IN {languagesAsQueryStr}");
+            }
+
+            if (authors != null && authors.Count > 0)
+            {
+                var authorsAsQueryStr = ArrayToQueryString<string>(authors);
+                queryStringBuilder.Append($" AND r.Author IN {authorsAsQueryStr}");
+            }
+
+            // Tags are ServiceNames for now.
+            if (tags != null && tags.Count > 0)
+            {
+                var tagsAsQueryStr = ArrayToQueryString<string>(tags);
+                queryStringBuilder.Append($" AND r.ServiceName IN {tagsAsQueryStr}");
+            }
+
+            if (isClosed != null)
+            {
+                queryStringBuilder.Append(" AND r.IsClosed = @isClosed");
+            }
+
+            if (filterTypes != null && filterTypes.Count > 0)
+            {
+                var filterTypeAsQueryStr = ArrayToQueryString<int>(filterTypes);
+                queryStringBuilder.Append($" AND r.FilterType IN {filterTypeAsQueryStr}");
+            }
+
+            if (isApproved != null)
+            {
+                queryStringBuilder.Append(" AND r.Revisions[0].IsApproved = @isApproved");
+            }
+
+            // First get the total count to help with paging
+            var countQuery = $"SELECT VALUE COUNT(1) FROM({queryStringBuilder.ToString()})";
+            QueryDefinition countQueryDefinition = new QueryDefinition(countQuery)
+                .WithParameter("@isClosed", isClosed)
+                .WithParameter("@filterTypes", filterTypes)
+                .WithParameter("@isApproved", isApproved);
+
+            using FeedIterator<int> countFeedIterator = _reviewsContainer.GetItemQueryIterator<int>(countQueryDefinition);
+            while (countFeedIterator.HasMoreResults)
+            {
+                result.TotalCount = (await countFeedIterator.ReadNextAsync()).SingleOrDefault();
+            }
+
+            queryStringBuilder.Append(" ORDER BY r.Name");

```

Comment:
Result should be ordered by modified timestamp.

Reply:
If we don't have LastUpdated property at review level then let's add it and sort the result using LastUpdated in descending order before splitting to pages. Otherwise first page won't show most recent items.